mailbox/postgres/src/main/java/org/apache/james/mailbox/jpa/mail/dao/PostgresMailboxDAOImpl.java

```diff
@@ -0,0 +1,165 @@
+            .onErrorMap(throwable -> throwable.getMessage().contains(DUPLICATE_VIOLATION_MESSAGE),
+                throwable -> new MailboxExistsException(mailbox.getName()));
+    }
+
+    @Override
+    public Mono<Void> delete(MailboxId mailboxId) {
+        return postgresExecutor.dslContext()
+            .flatMap(dslContext -> Mono.from(dslContext.deleteFrom(TABLE_NAME)
+                .where(MAILBOX_ID.eq(((PostgresMailboxId) mailboxId).asUuid()))))
+            .then();
+    }
+
+    @Override
+    public Mono<Mailbox> findMailboxByPath(MailboxPath mailboxPath) {
+        return postgresExecutor.dslContext()
+            .flatMapMany(dsl -> Flux.from(dsl.selectFrom(TABLE_NAME)
+                .where(MAILBOX_NAME.eq(mailboxPath.getName())
+                    .and(USER_NAME.eq(mailboxPath.getUser().asString()))
+                    .and(MAILBOX_NAMESPACE.eq(mailboxPath.getNamespace()))))
+                .map(this::generateMailbox))
+            .last()
+            .onErrorResume(NoSuchElementException.class, e -> Mono.empty());
+    }
+
+    @Override
+    public Mono<Mailbox> findMailboxById(MailboxId id) {
+        return postgresExecutor.dslContext()
+            .flatMapMany(dsl -> Flux.from(dsl.selectFrom(TABLE_NAME)
+                    .where(MAILBOX_ID.eq(((PostgresMailboxId) id).asUuid())))
+                .map(this::generateMailbox))
+            .last()
+            .onErrorResume(NoSuchElementException.class, e -> Mono.empty());
+    }
+
+    @Override
+    public Flux<Mailbox> findMailboxWithPathLike(MailboxQuery.UserBound query) {
+        String pathLike = MailboxExpressionBackwardCompatibility.getPathLike(query);
+
+        return postgresExecutor.dslContext()
+            .flatMapMany(dsl -> Flux.from(dsl.selectFrom(TABLE_NAME)
+                    .where(MAILBOX_NAME.like(pathLike)
+                        .and(USER_NAME.eq(query.getFixedUser().asString()))
+                        .and(MAILBOX_NAMESPACE.eq(query.getFixedNamespace()))))
+                .map(this::generateMailbox))
+            .filter(query::matches);
+    }
+
+    @Override
+    public Mono<Boolean> hasChildren(Mailbox mailbox, char delimiter) {
+        final String name = mailbox.getName() + delimiter + SQL_WILDCARD_CHAR;
+
+        return postgresExecutor.dslContext()
+            .flatMapMany(dsl -> Flux.from(dsl.select(count()).from(TABLE_NAME)
+                    .where(MAILBOX_NAME.like(name)
+                        .and(USER_NAME.eq(mailbox.getUser().asString()))
+                        .and(MAILBOX_NAMESPACE.eq(mailbox.getNamespace()))))
+                .map(Record1::value1)
+                .filter(count -> count > 0))
+            .hasElements();
+    }
+
+    @Override
+    public Flux<Mailbox> getAll() {
+        return postgresExecutor.dslContext()
+            .flatMapMany(dsl -> Flux.from(dsl.selectFrom(TABLE_NAME)))
+                .map(this::generateMailbox);
+    }
+
+    private Mailbox generateMailbox(Record record) {
```

Comment:
```suggestion
    private Mailbox asMailbox(Record record) {
```

We are not generating anything...